tests/core/test_otel_logger.py

```diff
@@ -356,3 +356,110 @@ def test_timer_start_and_stop_manually_send_true(self, mock_time, metrics_consis
         self.meter.get_meter().create_observable_gauge.assert_called_once_with(
             name=full_name(name), callbacks=ANY
         )
+
+    def test_incr_counter(self):
+        # Arrange
+        logger = self.stats
+        metric_name = "test_metric"
+        count = 5
+        tags = {"env": "prod"}
+
+        # Act
+        logger.incr(metric_name, count=count, tags=tags)
+
+        # Assert
+        counter = logger.metrics_map.get_counter(
+            full_name(prefix=logger.prefix, name=metric_name), attributes=tags
+        )
+        counter.add.assert_called_with(count, attributes=tags)
+
+    def test_decr_counter(self):
+        # Arrange
+        logger = self.stats
+        metric_name = "test_metric"
+        count = 3
+        tags = {"env": "prod"}
+
+        # Act
+        logger.decr(metric_name, count=count, tags=tags)
+
+        # Assert
+        counter = logger.metrics_map.get_counter(
+            full_name(prefix=logger.prefix, name=metric_name), attributes=tags
+        )
+        counter.add.assert_called_with(-count, attributes=tags)
+
+    @pytest.mark.parametrize("expected_duration", [2.5, 1.0, 3.0])
+    def test_timing(self, expected_duration):
+        # Arrange
+        logger = self.stats
+        metric_name = "test_metric"
+        tags = {"env": "prod"}
+
+        # Mocking time.perf_counter to simulate timing
+        with mock.patch.object(time, "perf_counter", side_effect=[0.0, expected_duration]):
+            with logger.timer(metric_name, tags=tags) as timer:
+                pass
+
+        # Assert
+        assert isinstance(timer.duration, float)
+        assert timer.duration >= expected_duration - 0.1
+        assert timer.duration <= expected_duration + 0.1
```

Comment:
Non-blocking Nitpick: `0.1` is a "magic number", consider creating a variable called "acceptable_margin" or something like that.